django/contrib/admin/widgets.py

```diff
@@ -382,3 +383,128 @@ def __init__(self, attrs=None):
+        return context
+
+    def build_attrs(self, base_attrs, extra_attrs=None):
+        """
+        Set select2's AJAX attributes.
+
+        Attributes can be set using the html5 data attribute.
+        Nested attributes require a double dash, please see:django/contrib/admin/checks.py
+        https://select2.github.io/options.html#how-can-i-set-a-default-value-for-a-nested-option
+        """
+        attrs = super().build_attrs(base_attrs, extra_attrs=extra_attrs)
+        attrs.setdefault('class', '')
+        attrs.update({
+            'data-ajax--cache': 'true',
+            'data-ajax--type': 'GET',
+            'data-theme': 'admin-autocomplete',
+            'data-allow-clear': json.dumps(not self.is_required),
+            'data-placeholder': '',  # Needs to be empty to allow clearing of the input
+            'class': attrs['class'] + 'admin-autocomplete',
+        })
+        return attrs
+
+    def get_field_name(self, name):
+        try:
+            # split prefix and name, reverse BaseForm.add_prefix()
+            return name.rsplit('-', 1)[-1]
+        except ValueError:
+            pass
+
+    def optgroups(self, name, value, attr=None):
+        """Return only selected options and set QuerySet from `ModelChoicesIterator`."""
+        default = (None, [], 0)
+        groups = [default]
+        has_selected = False
+        selected_choices = {str(v) for v in value}
+        if not self.is_required and not self.allow_multiple_selected:
+            default[1].append(self.create_option(name, '', '', False, 0))
+        selected_choices = {
+            c for c in selected_choices
+            if c not in self.choices.field.empty_values
+        }
+        choices = (
+            (obj.pk, self.choices.field.label_from_instance(obj))
+            for obj in self.choices.queryset.using(self.db).filter(pk__in=selected_choices)
+        )
+        for option_value, option_label in choices:
+            selected = (
+                str(option_value) in value and
+                (has_selected is False or self.allow_multiple_selected)
+            )
+            if selected is True and has_selected is False:
+                has_selected = True
+            index = len(default[1])
+            subgroup = default[1]
+            subgroup.append(self.create_option(name, option_value, option_label, selected_choices, index))
+        return groups
+
+    @property
+    def media(self):
+        available_translations = [
+            'ar', 'az', 'bg', 'ca', 'cs', 'da', 'de', 'el', 'en', 'es', 'et',
+            'eu', 'fa', 'fi', 'fr', 'gl', 'he', 'hi', 'hr', 'hu', 'id', 'is',
+            'it', 'ja', 'km', 'ko', 'lt', 'lv', 'mk', 'ms', 'nb', 'nl', 'pl',
+            'pt-BR', 'pt', 'ro', 'ru', 'sk', 'sr-Cyrl', 'sr', 'sv', 'th',
+            'tr', 'uk', 'vi', 'zh-CN', 'zh-TW',
+        ]
+        extra = '' if settings.DEBUG else '.min'
+        i18n_file = 'admin/js/vendor/select2/i18n/%s.js' % get_language()
```

Comment:
We could lowercase the vendored files, that would help at least for the `zh-*` variants.